### PR TITLE
Instance file push/pull/delete lifecycle events

### PR DESCRIPTION
**Pull request by masnax**
* add instance file lifecycle events for push, pull, and delete
* `InstanceFileRetrieved` includes the file source and destination as context
* `InstanceFilePushed` includes the file source, destination, and file info as context
* `InstanceFileRemoved` includes the file path as context

### CODE REVIEW

**Comment by stgraber** [comment]
@masnax needs a rebase

**Comment by masnax** [comment]
Oops :)

File: lxd/lifecycle/instance.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ const (
 	InstanceConsole       = InstanceAction("console")
 	InstanceFileRetrieved = InstanceAction("file-retrieved")
 	InstanceFilePushed    = InstanceAction("file-pushed")
+	InstanceFileRemoved   = InstanceAction("file-removed")
```
Comment (stgraber) [inline lxd/lifecycle/instance.go:39]: For consistency, I think we should actually make this `file-deleted`.

**Comment by stgraber** [comment]
@masnax aren't we also missing those 3 events for containers (driver_lxc)?

**Comment by masnax** [comment]
Huh, yeah I somehow overlooked that...

**Pull request merged**